README.md

```diff
@@ -34,6 +34,92 @@ ERROR: SystemError: opening file foo: No such file or directory
  in open at iostream.jl:99
 ```
 
+### @mendable
+
+Some functions you want to mock are called from the function you are testing. If the function you are testing is compiled, you will not be able to mock the internal function. That is where `@mendable` can help, it will allow you to mock the internal functions to your needs. As show in this example.
+
+```julia
+julia> function function_inside()
+           println("a")
+       end
+function_inside (generic function with 1 method)
+
+julia> test1() = function_inside()
+test1 (generic function with 1 method)
+
+julia> test2() = @mendable function_inside()
+test2 (generic function with 1 method)
+
+julia> test1()
+a
+
+julia> test2()
+a
+
+julia> function function_inside()
+           println("b")
+       end
+function_inside (generic function with 1 method)
+
+julia> test1()
+a
+
+julia> test2()
+b
+```
+
+### Patch
+
+`Patch` allows similar behaviour as you would get from using `mend` directly. Note that you only need to specify a signature when the function you are overloading has multiple methods and the replacement w ould be ambiguous. 
+
+```julia
+julia> replacement = (name::AbstractString) -> name == "foo" ? IOBuffer("bar") : Original.open(name)
+(anonymous function)
+
+julia> patches = [Patch(open, replacement)]
+1-element Array{Mocking.Patch,1}:
+ Mocking.Patch(open,(anonymous function),Mocking.Signature(Type[AbstractString]))
+
+julia> mend(patches) do
```

Comment:
I think `mend(Patch(open, replacement)) do` makes more since as we haven't introduced multiple patches yet.